### PR TITLE
Session endpoints

### PR DESCRIPTION
**Pull request by gazpachoking**
Renames some stuff to not confuse session 'stats' with 'status'.
Adds some endpoints to get/modify session settings.

### CODE REVIEW

**Review by aresch — CHANGES_REQUESTED** [review]
These are already implemented in /settings: https://github.com/spritzle/spritzle/blob/master/spritzle/resource/settings.py

Do you think it should be moved to /session/settings?

**Comment by gazpachoking** [comment]
Oh. Hmm, not sure how I missed that. Not sure now whether it should be moved.

**Comment by gazpachoking** [comment]
I think the old GET on /session giving the debug stats is a bit useless though.

**Comment by aresch** [comment]
I suppose we could move it to /session/stats .. I'm not really sure what we could return in /session though, maybe the output of session.save_state()?

**Comment by gazpachoking** [comment]
Well if the session settings stay in /settings, I thing the whole /session endpoint could go away? Are those stats that useful to a client?

Only slightly related to this discussion: If you change the session settings now, that won't be persisted through restarts, will it? Should it?

**Comment by aresch** [comment]
It should be persisted through session.save_state() which we write to disk and load on startup. I think keeping around the session stats could be useful, why remove it?

**Comment by gazpachoking** [comment]
Ahh, yeah, I think it may be some windows bullcrap that prevents the session save_state from calling for me (until there is a method to quit other than KeyboardInterrupt, which is not working on windows.) I don't really see a reason to remove the session stats, I just didn't have a strong desire to keep it. I'm fine with it staying in, doesn't harm anything.

**Comment by gazpachoking** [comment]
I'm thinking moving the settings up under /session makes sense, no idea what /session should return though.

File: spritzle/resource/session.py [hunk 1]
```diff
@@ -25,11 +25,36 @@
...
-async def get_session(request):
+@routes.get('/session/settings')
+async def get_session_settings(request):
     core = request.app['spritzle.core']
-    status = await core.get_session_status()
-    return web.json_response(status)
+    return web.json_response(core.session.get_settings())
+
+
+@routes.put('/settings')
```
Comment (aresch) [inline spritzle/resource/session.py]: @routes.put('/session/settings')

**Review by aresch — COMMENTED** [review]
The example for GET /session needs to be updated too.

**Comment by gazpachoking** [comment]
Ahh, good catch. Done.

**Comment by aresch** [comment]
I'll merge this PR, but you could please make a PR for spritzle-cli for the new endpoints?

**Review by aresch — APPROVED** [review]

**Comment by gazpachoking** [comment]
Yep, can do.

**Pull request merged**